bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/device/CommandHandler.java

```diff
@@ -0,0 +1,912 @@
+            }
+        }
+
+        private int getRampLevel(InsteonChannelConfiguration conf, int defaultValue) {
+            HashMap<String, @Nullable String> params = conf.getParameters();
+            return params.containsKey("ramplevel") ? Integer.parseInt(params.get("ramplevel")) : defaultValue;
+        }
+    }
+
+    @NonNullByDefault
+    public static class ManualChangeCommandHandler extends CommandHandler {
+        ManualChangeCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                if (cmd instanceof DecimalType) {
+                    int v = ((DecimalType) cmd).intValue();
+                    int cmd1 = (v != 1) ? 0x17 : 0x18; // start or stop
+                    int cmd2 = (v == 2) ? 0x01 : 0; // up or down
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, (byte) cmd1, (byte) cmd2, getGroup(conf));
+                    dev.enqueueMessage(m, feature);
+                    logger.debug("{}: cmd {} sent manual change {} {} to {}", nm(), v,
+                            (cmd1 == 0x17) ? "START" : "STOP", (cmd2 == 0x01) ? "UP" : "DOWN", dev.getAddress());
+                } else {
+                    logger.warn("{}: invalid command type: {}", nm(), cmd);
+                }
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
+            } catch (FieldException e) {
+                logger.warn("{}: command send message creation error ", nm(), e);
+            }
+        }
+    }
+
+    /**
+     * Sends ALLLink broadcast commands to group
+     */
+    @NonNullByDefault
+    public static class GroupBroadcastCommandHandler extends CommandHandler {
+        GroupBroadcastCommandHandler(DeviceFeature f) {
+            super(f);
+        }
+
+        @Override
+        public void handleCommand(InsteonChannelConfiguration conf, Command cmd, InsteonDevice dev) {
+            try {
+                if (cmd == OnOffType.ON || cmd == OnOffType.OFF) {
+                    byte cmd1 = (byte) ((cmd == OnOffType.ON) ? 0x11 : 0x13);
+                    byte value = (byte) ((cmd == OnOffType.ON) ? 0xFF : 0x00);
+                    int group = getGroup(conf);
+                    if (group == -1) {
+                        logger.warn("no group=xx specified in item {}", conf.getChannelName());
+                        return;
+                    }
+                    logger.debug("{}: sending {} broadcast to group {}", nm(), (cmd1 == 0x11) ? "ON" : "OFF",
+                            getGroup(conf));
+                    Msg m = dev.makeStandardMessage((byte) 0x0f, cmd1, value, group);
+                    dev.enqueueMessage(m, feature);
+                }
+            } catch (IOException e) {
+                logger.warn("{}: command send i/o error: ", nm(), e);
```

Comment:
This is actually here because Msg.makeMessage() throws an IOException if it can't find the message. This should never happen, except during development. Can I leave as is?

Reply:
done, created custom exception to replace IOException and left as warn.